### PR TITLE
feat: add device type's icons to cards

### PR DESCRIPTION
**Pull request by tbrkollar**

### CODE REVIEW

File: src/components/ZclEndpointCard.vue [hunk 1]
```diff
@@ -289,6 +300,15 @@ export default {
     }
   },
   methods: {
+    getDeviceCategory(packageRef) {
+      let category = ''
+      this.$store.state.zap.selectedZapConfig.zclProperties.forEach((item) => {
+        if (item.id === packageRef) {
+          category = item.category
```
Comment (brdandu) [inline src/components/ZclEndpointCard.vue:307]: `return category` once it has been populated.

File: src/util/common-mixin.js [hunk 2]
```diff
@@ -266,17 +266,30 @@ export default {
         (x) => this.sdkExtClusterCode(x) === clusterRoleName
       )
     },
+    createLogoSrc(isTiny, category, isSelected = false) {
+      return (
+        (isTiny ? '/logo/tiny/' : '/logo/') +
+        category +
```
Comment (brdandu) [inline src/util/common-mixin.js:272]: What happens when category is empty string (''). Some of our very old GSDKs did not have category.

File: src/util/common-mixin.js [hunk 3]
```diff
@@ -266,17 +266,30 @@ export default {
...
-                '.svg'
+              this.createLogoSrc(
+                this.$store.state.zap.isMultiConfig,
+                selectedZapConfig[i].category
+              )
+              // (this.$store.state.zap.isMultiConfig ? '/logo/tiny/' : '/logo/') +
+              //   selectedZapConfig[i].category +
+              //   '_logo' +
+              //   (this.$q.dark.isActive ? '_white' : '') +
+              //   '.svg'
```
Comment (brdandu) [inline src/util/common-mixin.js:292]: Get rid of comments

File: src/components/ZclEndpointCard.vue [hunk 4]
```diff
@@ -289,6 +300,15 @@ export default {
     }
   },
   methods: {
+    getDeviceCategory(packageRef) {
```
Comment (brdandu) [inline src/components/ZclEndpointCard.vue:303]: Add comments to function

File: src/util/common-mixin.js [hunk 5]
```diff
@@ -266,17 +266,30 @@ export default {
         (x) => this.sdkExtClusterCode(x) === clusterRoleName
       )
     },
+    createLogoSrc(isTiny, category, isSelected = false) {
```
Comment (brdandu) [inline src/util/common-mixin.js:269]: Add comments to function.

**Review by brdandu — CHANGES_REQUESTED** [review]

File: src/components/ZclEndpointCard.vue (hunk 4 above)
Comment (brdandu) [reply src/components/ZclEndpointCard.vue:304]: Keep the comment format as below for this function too
`/**
      *  Specifies the path to the logo image
      *
      * @param {*} isTiny -
      *                   if value "true", it's used as a tiny logo
      *                   if value "false", it's used as a normal logo
      * @param {*} category -
      *                   if value "zigbee", it's used as a zigbee logo
      *                   if value "matter", it's used as a matter logo
      *                   if value "multiprotocol", it's used as a multiprotocol logo
      * @param {*} isSelected -
      *                   if value "true", it's used as a dark / selected version
      *                   if value "false", it's used as a light version
      * @returns Returns a string value for the src image property
      */`

File: src/components/ZclEndpointCard.vue [hunk 6]
```diff
@@ -289,6 +300,19 @@ export default {
...
   },
   methods: {
+    // Calculate readable category string
+    getDeviceCategory(packageRef) {
+      let category = ''
+      this.$store.state.zap.selectedZapConfig.zclProperties.forEach((item) => {
+        if (item.id === packageRef) {
+          if (item.category) {
+            category = item.category
+            return
```
Comment (brdandu) [inline src/components/ZclEndpointCard.vue:310]: Just realized the returning in a forEach does not really end the loop but it continues going through the loop regardless.

Can you switch `this.$store.state.zap.selectedZapConfig.zclProperties.forEach((item) => {
         if (item.id === packageRef) {
           if (item.category) {
             category = item.category
             return
           }
         }
       })`
       
to

`
let zclProperty = this.$store.state.zap.selectedZapConfig.zclProperties.find((item) => (item.id === packageRef && item.category))
if (zclProperty) {
category = zclProperty.category
} 
return category
`

Comment (brdandu) [reply src/components/ZclEndpointCard.vue:310]: You can also switch to a normal for loop and return instead of using forEach

**Review by brdandu — APPROVED** [review]

File: src/components/ZclEndpointCard.vue (hunk 6 above)
Comment (tbrkollar) [reply src/components/ZclEndpointCard.vue:310]: you are right, thanks. i will fix it

**Pull request merged**